src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import static core.basesyntax.Name.RECTANGLE;
+
+public class Rectangle extends Figure {
+    private int length;
+    private int width;
+
+    public Rectangle(int length, int width) {
+        this.length = length;
+        this.width = width;
+        this.name = getName();
+        this.area = getArea();
+        this.color = getColor();
+    }
+
+    public int getLength() {
+        return length;
+    }
+
+    public int getWidth() {
+        return width;
+    }
+
+    @Override
+    public Name getName() {
+        return RECTANGLE;
+    }
+
+    @Override
+    public double getArea() {
+        return length * width;
+    }
+
+    @Override
+    public Color getColor() {
+        return new ColorSupplier().getRandomColor();
+    }
```

Comment:
With current logic your figure
1) Don't really have any color, it will be new each time you call getter
2) Creating new object of ColorSupplier what is very resources-wasting
```suggestion
```
ColorSupplier should be used in FigureSupplier to provide color for figure creation